warehouse/query-core/src/test/java/datawave/query/jexl/visitors/PushdownNegationVisitorTest.java

```diff
@@ -282,4 +283,15 @@ public void testApplyDeMorgansNegateOrRoot() {
         JexlNode result = PushdownNegationVisitor.applyDeMorgans(or, true);
         Assert.assertEquals("!((!(f1 == 'v1') && !(f2 == 'v2')))", JexlStringBuildingVisitor.buildQuery(result));
     }
+    
+    @Test
+    public void testPushdownNegationIntoContentFunction() throws ParseException {
+        String query = "FOO == 'few' && !(content:phrase('TEXT', termOffsetMap, 'bar', 'baz') && (TEXT == 'bar' && TEXT == 'baz'))";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        
+        JexlNode pushed = PushdownNegationVisitor.pushdownNegations(script);
+        String expected = "FOO == 'few' && ((!(content:phrase('TEXT', termOffsetMap, 'bar', 'baz')) || !(TEXT == 'bar') || !(TEXT == 'baz')))";
```

Comment:
I feel like I'm missing something here, but I'll ask the question anyway: The first form enforces that the query does not match "bar baz", "bar" and "baz". The second form seems to convert this to not match "bar baz" or not match "bar" or not match "baz". 

The user's intent in the first form is to exclude "bar baz" from the results, but it seems like the second form would exclude "bar" and/or "baz" if they appear by themselves outside the context of a phrase.

Reply:
This is a fun one. The first form says that both 'bar' and 'baz' must be present to evaluate the function. If either term is missing then the entire content function cannot be evaluated, which in turn evaluates that branch of the query to true (candidate document was *not* excluded).

In the second form we need the second part to evaluate to true (!content:function || !termA || !termB). The absence of term B will force the entire subtree to a True. The presence of term B will force (NOT TRUE) into false, and we continue evaluating the union. So it should all work out in the end.